src/main/java/com/minecolonies/coremod/client/render/worldevent/ItemOverlayBoxesRenderer.java

```diff
@@ -0,0 +1,53 @@
+package com.minecolonies.coremod.client.render.worldevent;
+
+import com.ldtteam.structurize.util.WorldRenderMacros;
+import com.minecolonies.api.items.IBlockOverlayItem;
+import com.mojang.blaze3d.vertex.VertexConsumer;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.util.FastColor;
+
+import java.util.List;
+
+/**
+ * Renders boxes for {@link com.minecolonies.api.items.IBlockOverlayItem}
+ */
+public class ItemOverlayBoxesRenderer
+{
+    /**
+     * Renders overlay boxes into the client.
+     *
+     * @param ctx rendering context
+     */
+    static void render(final WorldEventContext ctx)
+    {
+        if (ctx.mainHandItem.getItem() instanceof final IBlockOverlayItem overlayItem)
+        {
+            final List<IBlockOverlayItem.OverlayBox> boxes = overlayItem.getOverlayBoxes(ctx.clientLevel, ctx.clientPlayer, ctx.mainHandItem);
+
+            for (final IBlockOverlayItem.OverlayBox box : boxes)
+            {
+                final RenderType renderType = box.clipped() ? WorldRenderMacros.LINES_WITH_WIDTH : WorldRenderMacros.GLINT_LINES_WITH_WIDTH;
+                final VertexConsumer buffer = ctx.bufferSource.getBuffer(renderType);
+
+                final float halfLine = box.width() / 2.0f;

```

Comment:
There is renderLineBox which you have probably copy from, why not use that one?

Reply:
Because that only supports `BlockPos` coords, not an `AABB` which allows smaller-than-block positions.  While granted in these particular cases I'm only drawing boxes around entire blocks, I liked the idea of allowing smaller boxes when that seemed useful.